src/Quaternion.jl

```diff
@@ -29,14 +39,72 @@ quat(p, v1, v2, v3, n) = Quaternion(p, v1, v2, v3, n)
 quat(x) = Quaternion(x)
 quat(s, a) = Quaternion(s, a)
 
+"""
+    real(T::Type)
+
+Return the type that represents the real part of a value of type `T`.
+e.g: for `T == Quaternion{R}`, returns `R`.
+Equivalent to `typeof(real(zero(T)))`.
+
+# Examples
+```jldoctest
+julia> real(Quaternion{Int})
+Int64
+```
+"""
 real(::Type{Quaternion{T}}) where {T} = T
+
+"""
+    real(q::Quaternion)
+
+Return the real part of the quaternion `q`.
+
+See also: [`imag_part`](@ref), [`quat`](@ref), [`isreal`](@ref).
+
+# Examples
+```jldoctest
+julia> real(quat(1,2,3,4))
+1
+
+julia> real([quat(5,6,7,8), 9])
+2-element Vector{Int64}:
+ 5
+ 9
+```
+"""
 real(q::Quaternion) = q.s
+
```

Comment:
This is how we can document `real(::AbstractArray{<:Quaternion})` without defining a method:

```suggestion

"""
    real(A::AbstractArray{<:Quaternion})
    
Return an array containing the real part of each quaternion in `A`.

# Examples
````jldoctest
julia> real([quat(5,6,7,8), 9])
2-element Vector{Int64}:
 5
 9
````
"""
Base.real(::AbstractArray{<:Quaternion})

```

Reply:
I didn't know that. Thanks!